### PR TITLE
Bump to vanilla-4.7.0

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Bump to [vanilla-4.7.0](https://vanillaframework.io/docs/whats-new)
- removes an unused class

## QA

- Check that nothing is broken after upgrading vanilla

### CODE REVIEW

File: static/sass/styles.scss [hunk 1]
```diff
@@ -1366,20 +1366,6 @@ $color-link-dark: #69c !default;
   }
 }
 
-// XXX: @bartaz: this will be covered in Vanilla
-// can be removed once https://github.com/canonical/vanilla-framework/issues/4875 is done and released
-.p-logo-section.has-misaligned-images {
```
Comment (mtruj013) [inline static/sass/styles.scss:1371]: I haven't finished qa'ing all the pages that use this, but it looks like the logo sections in https://ubuntu-com-13548.demos.haus/financial-services still looks buggy without it compare to prod

**Review by mtruj013 — COMMENTED** [review]
Couple meganav bugs:

bubble nav items aren't aligned as expected: 
Screenshot from 2024-02-07 16-08-26
vs: 
Screenshot from 2024-02-07 16-08-33

Search icon appears on new line on landing: 
Screenshot from 2024-02-07 16-06-04

**Comment by bartaz** [comment]
> Couple meganav bugs:

@petesfrench @mtruj013 The main change in 4.7.0 was the increased padding of top navigation items. With 4.7.0 top nav items have padding equal to grid margin width (on large screens).
So indeed it may cause some issues there. If anything was aligned to nav before, this will need to be realigned.

Similarly if there is a lot of elements in the nav (and meganav seems to be very packed) the $breakpoint-navigation-threshold may need to be adjusted to drop to mobile view as soon as they don't fit.

If there are any issues that make the new padding too large for meganav ping me and @lyubomir-popov, as maybe we would need some exception for such dense navigation…

**Comment by petesfrench** [comment]
@bartaz @lyubomir-popov In the case of the meganav we already use a custom `padding-right: 2.5rem` to account for the down arrow and the additional `0.5rem` on each button will make it too wide. In this situation would it be viable to maintain the `padding-left: 1rem` that we are currently using?

**Comment by bartaz** [comment]
Also, 4.7.0 introduces `p-navigation__row--25-75` that is more reliable way to implement grid in top nav than using `row` and cols.

https://vanillaframework.io/docs/examples/patterns/navigation/default

File: static/sass/_pattern_navigation.scss [hunk 2]
```diff
@@ -200,6 +200,10 @@ $meganav-height: 3rem;
     }
 
     .p-navigation__row--25-75 {
+      .p-navigation__items:first-child {
```
Comment (bartaz) [inline static/sass/_pattern_navigation.scss:203]: Why is this needed? `p-navigation__row--25-75` was added specifically to avoid need for custom styling to match the grid.

Comment (bartaz) [reply static/sass/_pattern_navigation.scss:203]: Ah, ok, it's because you override the left padding on the nav items, so you need to compensate.

File: templates/templates/_navigation.html [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {% set breadcrumbs = get_navigation(request.path).breadcrumbs %}
 <header id="navigation" class="p-navigation--sliding is-dark {% if breadcrumbs and breadcrumbs.children or breadcrumbs.grandchildren %}is-reduced {% elif breadcrumbs == 'tutorials' %}is-reduced{% endif %}">
   {% block header_banner %}
-  <div class="row">
-    <div class="p-navigation__banner col-3">
+  <div class="p-navigation__row--25-75">
+    <div class="p-navigation__banner col">
```
Comment (bartaz) [inline templates/templates/_navigation.html:5]: `.col` is not needed, as the navigation always consists of the __banner and __nav, they are targeted directly.

```suggestion
    <div class="p-navigation__banner">
```

File: templates/templates/_navigation.html [hunk 4]
```diff
@@ -27,7 +27,7 @@
         </li>
       </ul>
     </div>
-    <nav class="p-navigation__nav col-9 js-show-nav" aria-label="Categories">
+    <nav class="p-navigation__nav col js-show-nav" aria-label="Categories">
```
Comment (bartaz) [inline templates/templates/_navigation.html:30]:
```suggestion
    <nav class="p-navigation__nav js-show-nav" aria-label="Categories">
```

File: templates/templates/_navigation.html [hunk 5]
```diff
@@ -75,8 +75,8 @@
 {% if not(hide_subnav == True) %}
 {% if breadcrumbs and breadcrumbs.children or breadcrumbs.grandchildren %}
 <div id="secondary-navigation" class="p-navigation is-secondary is-dark">
-  <div class="row">
-    <div class="p-navigation__banner col-3">
+  <div class="p-navigation__row--25-75">
+    <div class="p-navigation__banner col">
```
Comment (bartaz) [inline templates/templates/_navigation.html:79]:
```suggestion
    <div class="p-navigation__banner">
```

File: templates/templates/_navigation.html [hunk 6]
```diff
@@ -92,7 +92,7 @@
       {% endif %}
     </div>
 
-    <nav class="p-navigation__nav col-9" aria-label="{{ breadcrumbs.section.title | replace(' ', '&nbsp;') | safe }} navigation">
+    <nav class="p-navigation__nav col" aria-label="{{ breadcrumbs.section.title | replace(' ', '&nbsp;') | safe }} navigation">
```
Comment (bartaz) [inline templates/templates/_navigation.html:95]:
```suggestion
    <nav class="p-navigation__nav" aria-label="{{ breadcrumbs.section.title | replace(' ', '&nbsp;') | safe }} navigation">
```

File: static/sass/_pattern_navigation.scss [hunk 7]
```diff
@@ -53,14 +53,22 @@ $meganav-height: 3rem;
...
     .p-navigation__nav {
-      margin-left: -1rem;
       margin-right: 0;
       max-height: calc(100vh - 3rem);
       width: auto;
 
       .p-navigation__items {
         width: 100%;
 
+        & > .p-navigation__item--dropdown-toggle > .p-navigation__link {
```
Comment (bartaz) [inline static/sass/_pattern_navigation.scss:63]: It's probably out of scope for this PR, as it doesn't originate here, but the way the navigation is implemented here, with A LOT of overrides of very specific Vanilla class names is quite fragile and possibly problematic whenever we update something in Vanilla.

Variants like these ideally should be implemented with different class names from Vanilla. So instead of overriding `& > .p-navigation__item--dropdown-toggle > .p-navigation__link`, you would define `p-navigation__link--dense` (or whatever is the purpose of this) and use this new class name instead.

But as I mentioned, it's not just this one thing in this PR, the whole navigation seems to be written this way, so I mention this as maybe a potential for future improvements/refactoring.

**Review by bartaz — APPROVED** [review]
Code looks good to me, but I don't know which parts are relevant for QA to make sure there are no regressions.

File: static/sass/styles.scss (hunk 1 above)
Comment (bartaz) [reply static/sass/styles.scss:1371]: Also, this seems to just remove the comment, but not actual code. Is this intended?

File: static/sass/_pattern_navigation.scss (hunk 7 above)
Comment (petesfrench) [reply static/sass/_pattern_navigation.scss:63]: Ok, I have made a separate issue to tackle this as not to block this work https://warthogs.atlassian.net/browse/WD-8935

**Comment by juanruitina** [comment]
I see the search button have some padding that didn't before. This might seem minor, but it is reducing the gap between "Get Ubuntu" and "All Canonical", making it barely noticeable particularly when the user is logged in.

There are some alignment issues on the mobile navigation, at the left of the nav items:

ubuntu-com-13548 demos haus_desktop(iPhone SE)
ubuntu-com-13548 demos haus_(iPhone SE)

**Comment by petesfrench** [comment]
@lyubomir-popov could you give this a look please

**Comment by juanruitina** [comment]
> I see the search button have some padding that didn't before. This might seem minor, but it is reducing the gap between "Get Ubuntu" and "All Canonical", making it barely noticeable particularly when the user is logged in.

Elaborating a bit on this, see these screenshots:

This is what it looks like in the live version:

ubuntu com_

This is what it looks like in the demo:

ubuntu-com-13548 demos haus_

See how the gap between "Get Ubuntu" and "All Canonical" virtually disappears, making it all look like one long list instead of two, one left aligned the other right aligned. This arguably increases cognitive download and hinders the sense of hierarchy that separating them provides. @lyubomir-popov thoughts on it?

**Comment by bartaz** [comment]
@juanruitina @petesfrench There seems to be some additional padding in this PR on the right.

What is life seems correct: search button (when hovered) should end exactly where grid ends:

image

On the demo there is some additional "padding" that squeezes the navigation:

image

But I can't figure out what causes this. It's not actuall padding or margin, it seems like the items in the nav are not spanning whole width of available space.

**Comment by bartaz** [comment]
@petesfrench I think I see what the issue is.

You have all the nav items in a single `p-navigation__items` list. 

But if you want to have 2 sets: one left aligned, and the other right aligned, you need two separate `p-navigation__items` lists, not one.

Compare to Vanilla home page:

image

https://vanillaframework.io/

The misalignment is caused because the first navigation items list is shifted to the left (to align with the grid). So if it includes the right aligned items, those are shifted as well. Right aligned items need to be in separate list.

**Comment by petesfrench** [comment]
@bartaz @juanruitina I have added a negative margin to the top nav so it aligns. The reason for having them all in my list is to simplify the JS logic that handles the navigation. Sticking to the vanilla pattern would be better but would require some refactoring of the code which I would like to avoid right now. What do you think?

**Comment by petesfrench** [comment]
Merging without 4.8.0 as it requires more SCSS changes and would grow the scope of this PR.
